reactive/etcd.py

```diff
@@ -329,6 +329,8 @@ def install_etcd():
     if channel:
         snap.install('etcd', channel=channel, classic=False)
 
+    set_state('etcd.installed')
```

Comment:
Setting `etcd.installed` here will confuse the charm and make it think that a migration to snaps needs to occur, won't it?

`etcd.installed` is a legacy state that comes from the old days when we delivered etcd via charm resource, not snap. The charm still uses that state as a signal that the upgrade from pre-snap -> snap has not occurred. See for example this handler:

https://github.com/charmed-kubernetes/layer-etcd/blob/0e450a360598f4e4ac247e343e05b9aba9e1f04f/reactive/etcd.py#L83-L85